src/lib/provable/test/dynamic-array.unit-test.ts

```diff
@@ -0,0 +1,312 @@
+import { UInt8, Field, ZkProgram } from 'o1js';
+import { DynamicArray } from '../dynamic-array.js';
+import { assert } from '../gadgets/common.js';
+
+function expectThrows(fn: () => void | Promise<void>, msg: string) {
+  let threw = false;
+  try {
+    fn();
+  } catch {
+    threw = true;
+  }
+  assert(threw, msg);
+}
+
+let List = ZkProgram({
+  name: 'dynamicarrays',
+  methods: {
+    pushAndPop: {
+      privateInputs: [],
+      async method() {
+        // Define classes of dynamic arrays for specific provable types
+        class Bytestring extends DynamicArray(UInt8, { capacity: 8 }) {}
+
+        // Initialize an empty dynamic array
+        let bytes = new Bytestring();
+
+        // Initialization should produce right capacity and length
+        assert(bytes.capacity === 8);
```

Comment:
We should move these out-of-circuit tests elsewhere, they aren't valid provable code